test/pkg/environment/common/setup.go

```diff
@@ -113,7 +132,9 @@ func (env *Environment) Cleanup() {
 
 func (env *Environment) AfterEach() {
 	debug.AfterEach(env.Context)
-	env.printControllerLogs(&corev1.PodLogOptions{Container: "controller"})
+	if env.InClusterController {
```

Comment:
`env.ExpectNoCrashes()` above should also be conditional on `env.InClusterController`. It will just silently fail to detect crashes; best to have a clear indication this expectation is not checked.

Reply:
Addressed